src/diffChecker.ts

```diff
@@ -29,44 +30,57 @@ export const diffChecker = (
   const nonZeroTest = (x: number) => x !== 0;
   const coverageDecreased = (x: number) =>
     x < 0 ? Math.abs(x) >= coverageDecreaseThreshold : false;
-  const isBelowThreshold = (x: number) => x < coverageThreshold;
+    const isBelowThreshold = (x: number) => x < coverageThreshold;
+    const isBelowNewFileThreshold = (x: number) => x < newFileCoverageThreshold;
 
   diffMap.forEach((v, k) => {
     const diffPercentages = getSummaryPercentages(v);
 
-    // Only check files that changed coverage.
-    if (Object.values(diffPercentages).some(nonZeroTest)) {
-      const decreased = checkCoverageForCondition(
+    let itemBelowThreshold = false;
+    let decreased = false;
+
+    // Only check files that changed coverage. or new files
+    if (v.isNewFile) {
+      itemBelowThreshold = checkCoverageForCondition(
+        head[k],
+        checkCriteria,
+        isBelowNewFileThreshold
+      );
+    } else if (Object.values(diffPercentages).some(nonZeroTest)) {
+      decreased = checkCoverageForCondition(
         v,
         checkCriteria,
         coverageDecreased
       );
-      const itemBelowThreshold = checkCoverageForCondition(
+      itemBelowThreshold = checkCoverageForCondition(
         head[k],
         checkCriteria,
         isBelowThreshold
       );
+    } else {
+      return;
```

Comment:
An early exit can be ok if it's at the top of the code block but in this case being somewhere in the middle it can turn into a trap.

With the idea above we would also return to only one case (the if where we "Only check files that changed coverage.") and get rid of this early loop exit and only do "work" if needed.

Reply:
See my new update, I'm using an early return as a guard statement, I think it pairs well with the new structure of this function.